sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -68,9 +68,6 @@ object PhysicalOperation extends PredicateHelper {
         val substitutedCondition = substitute(aliases)(condition)
         (fields, filters ++ splitConjunctivePredicates(substitutedCondition), other, aliases)
 
-      case h: ResolvedHint =>
-        collectProjectsAndFilters(h.child)
-
```

Comment:
Hi, @jzhuge .
Do we need this? According to the previous PR, `master` has no issue. So, I expected a test only PR.

cc @cloud-fan and @gatorsmile

Reply:
Yeah, it is safer to only port the unit test. However, we believe this is dead code, otherwise we might have to revisit the fix for 2.4 and 2.3.